aikau/src/main/resources/alfresco/layout/DynamicVisibilityResizingMixin.js

```diff
@@ -0,0 +1,87 @@
+ * and the [subscribeToVisibilityRuleTopics]{@link module:alfresco/layout/DynamicVisibilityResizingMixin#subscribeToVisibilityRuleTopics}
+ * function should be called after widget processing has been completed (i.e. in an extension to 
+ * [allWidgetsProcessed]{@link module:alfresco/core/CoreWidgetProcessing#allWidgetsProcessed}).
+ * 
+ * @module alfresco/layout/DynamicVisibilityResizingMixin
+ * @author Dave Draper
+ * @since 1.0.38
+ */
+define(["dojo/_base/declare",
+        "dojo/_base/lang",
+        "dojo/_base/array"], 
+        function(declare, lang, array) {
+   
+   return declare([], {
+      
+      /**
+       * This array is setup when the [getVisibilityRuleTopics]{@link module:alfresco/layout/DynamicVisibilityResizingMixin#getVisibilityRuleTopics}
+       * is called and each topic is then subscribed to in order to trigger resize events when widgets are displayed or hidden.
+       *
+       * @instance
+       * @type {string[]}
+       * @default
+       */
+      visibilityRuleTopics: null,
+
+      /**
+       * This function can be called to check all the supplied widgets for dynamic visibility configuration so that 
+       * subscriptions can be created on the same rules to trigger resizing as widgets are displayed or hidden.
+       * 
+       * @instance
+       * @param {object[]} widgets The widgets to check for visibility/invisibility configuration
+       * @return {string[]} An array of the topics that are using in dynamic visibility/invisibility configuration
+       */
+      getVisibilityRuleTopics: function alfresco_layout_DynamicVisibilityResizingMixin__getVisibilityRuleTopics(widgets) {
+         var topicNames = {};
+         array.forEach(widgets, function(widget) {
+            var visibilityRules = lang.getObject("config.visibilityConfig.rules", false, widget) || [];
+            var invisibilityRules = lang.getObject("config.invisibilityConfig.rules", false, widget) || [];
+            array.forEach(visibilityRules.concat(invisibilityRules), function(rule) {
+               if (rule.topic) {
+                  topicNames[rule.topic] = true;
+               }
+            });
+         });
+         return Object.keys(topicNames);
+      },
+
+      /**
+       * Iterates over the [visibilityRuleTopics]{@link module:alfresco/layout/DynamicVisibilityResizingMixin#visibilityRuleTopics}
+       * and creates a subscription for each one bound to the supplied function.
+       * 
+       * @instance
+       * @param {function} f The function to bind each subscription to.
+       */
+      subscribeToVisibilityRuleTopics: function alfresco_layout_DynamicVisibilityResizingMixin__subscribeToVisibilityRuleTopics(f) {
+         if (this.visibilityRuleTopics)
+         {
+            array.forEach(this.visibilityRuleTopics, function(topic) {
+               this.alfSubscribe(topic, lang.hitch(this, f));
```

Comment:
Single-letter variables are really unhelpful when trying to understand what's going on, as you have to look at the original declaration, and hope that it will give enough information to see what it is. Perhaps this could be renamed `subFunc` or similar?